src/EventStore.Core/Services/Transport/Grpc/RpcExceptions.cs

```diff
@@ -76,7 +76,7 @@ public static class RpcExceptions {
 			new(
 				new Status(
 					StatusCode.FailedPrecondition,
-					$"Append failed due to WrongExpectedVersion. Stream: {streamName}, Expected version: {expectedVersion}, Actual version: {actualVersion}"),
+					$"Delete failed due to WrongExpectedVersion. Stream: {streamName}, Expected version: {expectedVersion}, Actual version: {actualVersion}"),
```

Comment:
This changes the `RpcException.WrongExpectedVersion` method to be specific to the `Delete` operation, but there's nothing in the method signature that makes this obvious. While delete is currently the only operation using this, if someone were to add another operation that returns a `WrongExpectedVersion` exception then they may use this without realising that it'll return the wrong error message.

You could pass in an operation name to use in the error message (my preference), or perhaps change the method name to reflect that it's specific for `Delete`.

@timothycoleman what do you think?